tests/test_codegen.py

```diff
@@ -38,10 +38,31 @@ def test_bridge_namespaces(from_namespace, to_namespace, name, expected):
 
 def test_populate_namespaces_produces_roundtrippable_modules():
     populated_schemas = codegen.populate_namespaces(
-        parse_into_types(schema)
-        for schema in TEST_SCHEMAS
+        [parse_into_types(schema) for schema in TEST_SCHEMAS]
     )
 
     for _, module in populated_schemas:
         source = astor.to_source(module)
         assert isinstance(ast.parse(source), ast.Module)
+
+
+def test_schema_reference_resolution():
+    parsed_schemas = [parse_into_types(schema) for schema in TEST_SCHEMAS]
```

Comment:
This test needs a docstring explaining what properties it's actually trying to ensure hold.